main.go

```diff
@@ -66,53 +72,117 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
-	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
-	metrics.Registry.MustRegister(common_templates.CommonTemplatesRestored)
-	metrics.Registry.MustRegister(common.SSPOperatorReconcileSucceeded)
-	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
-	mux := http.NewServeMux()
-	mux.Handle("/metrics", handler)
+// This callback executes on each client call returning a new config to be used
+// please be aware that the APIServer is using http keepalive so this is going to
+// be executed only after a while for fresh connections and not on existing ones
+func getConfigForClientCallback(cfg *tls.Config, apiClient client.Client, ctx context.Context) (*tls.Config, error) {
```

Comment:
Three comments for this function:
- After thinking about this more, I think it is better to have a parameter type `cache.Cache` instead of `client.Client`. It has the same reading methods as `Client`, but cannot be used for writing.
- Usually the `context.Context` parameter is the first.
- Please don't include the word `Callback` in a function name. In my opinion, it's better to name a function, by what it does, then by how it is used.